pkg/plugins/backendplugin/provider/provider.go

```diff
@@ -68,9 +65,6 @@ var SecretsManagerProvider PluginBackendProvider = func(_ context.Context, p *pl
 	)
 }
 
-func DefaultProvider(features featuremgmt.FeatureToggles) PluginBackendProvider {
-	return func(_ context.Context, p *plugins.Plugin) backendplugin.PluginFactoryFunc {
-		skipEnvVars := features.IsEnabledGlobally(featuremgmt.FlagPluginsSkipHostEnvVars)
-		return grpcplugin.NewBackendPlugin(p.ID, p.ExecutablePath(), skipEnvVars)
-	}
-}
+var DefaultProvider = PluginBackendProvider(func(_ context.Context, p *plugins.Plugin) backendplugin.PluginFactoryFunc {
+	return grpcplugin.NewBackendPlugin(p.ID, p.ExecutablePath(), p.SkipHostEnvVars)
+})
```

Comment:
This is the same signature as before #77858. We don't need to access feature flags in there anymore :sweat_smile: , because I introduced `plugin.SkipHostEnvVars`